packages/email-petition/src/email-petition.tsx

```diff
@@ -183,16 +198,99 @@ export function EmailPetition(props: {
 
   const outsideUS = watch("outsideUS");
   const zip = watch("zip");
-  const isInSonomaCounty = useMemo(() => {
-    return zip && zip in SonomaCities;
-  }, [zip]);
-  const cities = useMemo(() => {
-    if (!isInSonomaCounty) {
-      return [];
+
+  // Fetch postcode_localities from Google Geocoding API
+  async function fetchPostcodeLocalities(zipcode: string) {
+    try {
+      const response = await ky
+        .get(
+          `https://maps.googleapis.com/maps/api/geocode/json?address=${zipcode}&key=${props.areaScope.googleGeocodingApiKey}`,
+        )
+        .json<{ results: any[] }>();
+
+      const scopeComponentType = (() => {
+        switch (props.areaScope.subdivision) {
+          case "city":
+            return "locality";
+          case "county":
+            return "administrative_area_level_2";
+          case "state":
+            return "administrative_area_level_1";
+          case "country":
+            return "country";
+        }
+      })();
+
+      const scopeName = props.areaScope.name;
+      const scopeNameLower = scopeName.toLowerCase();
+
+      const scopedResults = (response.results || []).filter((result) =>
+        (result.address_components || []).some((component: any) => {
+          if (!component?.types?.includes(scopeComponentType)) {
+            return false;
+          }
+          return (
+            String(component.long_name).toLowerCase() === scopeNameLower ||
+            String(component.short_name).toLowerCase() === scopeNameLower
+          );
+        }),
+      );
+
+      const resultsToUse = scopedResults.length ? scopedResults : [];
+
+      //if there are multiple cities/localities per zipcode
+      const postcodeLocalities = resultsToUse
+        .filter((result) => result.types?.includes("postal_code"))
+        .flatMap((result) => result.postcode_localities || []);
```

Comment:
do we expect postcode_localities to be falsy given the `.filter` call above? if not, the `|| []` at the end of this line will just obscure the error. perhaps we can assert that it is not null/undefined instead so any error will be thrown closer to the source of the issue.